custom/icds_reports/utils/aggregation_helpers/distributed/agg_service_delivery_report.py

```diff
@@ -136,36 +154,36 @@ def update_queries(self):
                 SUM(CASE WHEN thr_eligible=1 AND num_rations_distributed>=21 THEN 1 ELSE 0 END) as mother_thr_21_days
             FROM ccs_record_monthly
             WHERE month=%(start_date)s
-            GROUP BY supervisor_id, awc_id
+            GROUP BY supervisor_id, awc_id, month
+        ) ut
+        WHERE (
+            agg_sdr.awc_id=ut.awc_id AND
+            agg_sdr.aggregation_level=5 AND
+            agg_sdr.month=ut.month
+        )
         """,{
             "start_date":self.month
         }
 
-        yield """
-        UPDATE "{tmp_tablename}" agg_sdr
-        SET thr_eligible = thr_eligible +  ut.mother_thr_eligible,
-            thr_0_days = thr_0_days +  ut.mother_thr_0_days,
-            thr_1_7_days = thr_1_7_days +  ut.mother_thr_1_7_days,
-            thr_8_14_days = thr_8_14_days +  ut.mother_thr_8_14_days,
-            thr_15_20_days = thr_15_20_days +  ut.mother_thr_15_20_days,
-            thr_21_days = thr_21_days +  ut.mother_thr_21_days
-        FROM temp_ccs_table ut
-        WHERE agg_sdr.awc_id = ut.awc_id;
-        """.format(
-            tmp_tablename=self.staging_tablename,
-        ), {}
-
-        yield """
-        DROP TABLE temp_ccs_table;
-        """, {}
+    def update_queries(self):
+        yield f"""
```

Comment:
it might be nice to separate these into 3 queries executed separately so that its easier to log. also, as before, would be good to drop the table at the start in case the last run failed to clean up correctly